src/Sylius/Bundle/LocaleBundle/Templating/Helper/LocaleHelper.php

```diff
@@ -41,6 +42,188 @@ public function getCurrentLocale()
     }
 
     /**
+     * @return string
+     */
+    public function getCurrentCalendar()
+    {
+        return $this->localeContext->getCurrentCalendar();
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrentDirection()
+    {
+        return $this->localeContext->getCurrentDirection();
+    }
+
+    /**
+     * Format date in a calendar-aware fashion
+     *
+     * @param string|\DateTime     $date     Date to format
+     * @param string               $format   PHP-compatible date format
+     * @param string|\DateTimeZone $timezone Timezone
+     * @param string               $calendar Calendar name
+     *
+     * @return bool|string Calendar-aware localized formatted date
+     */
+    public function formatDate($date, $format = null, $timezone = null, $locale = null, $calendar = null)
+    {
+        // Determine the timezone
+        if (!$timezone) {
+            $timezone = date_default_timezone_get();
+        }
+
+        if (!$timezone instanceof \DateTimeZone) {
+            $timezone = new \DateTimeZone($timezone);
+        }
+
+        if ($locale === null) {
+            $locale = $this->localeContext->getCurrentLocale();
+        }
+
+        if ($calendar === null) {
+            $calendar = $this->localeContext->getCurrentCalendar();
+        }
+
+        if ($date instanceof \DateTime || $date instanceof \DateTimeInterface) {
+            $date = clone $date;
+
+            if (false !== $timezone) {
+                $date->setTimezone($timezone);
+            }
+        } else {
```

Comment:
Another thing is, could you extract this logic to separate private method and use early return statement instead of else?